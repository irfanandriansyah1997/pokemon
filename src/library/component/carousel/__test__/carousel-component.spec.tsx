import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Carousel from '@/library/component/carousel';
import {
  CarouselIndicator,
  CarouselNext,
  CarouselPrev
} from '@/library/component/carousel/style';

describe(`Testing Carousel Component`, () => {
  it(`Testing Snapshot Component`, () => {
    const ui = shallow(
      <Carousel item={[`image-1.jpg`, `image-2.jpg`]} showCarousel />
    );

    expect(toJson(ui)).toMatchSnapshot();
  });

  describe(`Testing Interact Carousel`, () => {
    it(`Simulate Click Next / Previous`, () => {
      const ui = mount(
        <Carousel item={[`image-1.jpg`, `image-2.jpg`]} showCarousel />
      );

      expect(ui.find(CarouselIndicator).at(0).prop(`indicator`)).toBe(0);
      expect(ui.find(`div div`).at(0).prop(`style`)).toStrictEqual({
        transform: `translateX(0%)`
      });

      ui.find(CarouselNext).at(0).simulate(`click`);
      ui.update();

      expect(ui.find(CarouselIndicator).at(0).prop(`indicator`)).toBe(1);
      expect(ui.find(`div div`).at(0).prop(`style`)).toStrictEqual({
        transform: `translateX(-100%)`
      });

      ui.find(CarouselPrev).at(0).simulate(`click`);
      ui.update();

      expect(ui.find(CarouselIndicator).at(0).prop(`indicator`)).toBe(0);
      expect(ui.find(`div div`).at(0).prop(`style`)).toStrictEqual({
        transform: `translateX(0%)`
      });
    });
  });
});
