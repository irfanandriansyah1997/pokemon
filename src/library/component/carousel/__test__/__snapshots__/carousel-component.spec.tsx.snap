// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Carousel Component Testing Snapshot Component 1`] = `
<Styled(div)
  showCarousel={true}
>
  <div
    style={
      Object {
        "transform": "translateX(0%)",
      }
    }
  >
    <Styled(div)
      key="image-1.jpg"
    >
      <img
        alt="Pokemon"
        loading="lazy"
        src="image-1.jpg"
      />
    </Styled(div)>
    <Styled(div)
      key="image-2.jpg"
    >
      <img
        alt="Pokemon"
        loading="lazy"
        src="image-2.jpg"
      />
    </Styled(div)>
  </div>
  <Styled(div)
    onClick={[Function]}
    show={false}
  />
  <Styled(div)
    onClick={[Function]}
    show={true}
  />
  <Styled(div)
    indicator={0}
  >
    <div />
    <div />
  </Styled(div)>
</Styled(div)>
`;
